minahilfaisal/edx-platform
# Organization Field - Added to Registration Form API

## Description

This pull request adds a new "Organization" field to the registration form for the `edx-platform/lms` backend, allowing users to specify their affiliated organization during the registration process.

This dropdown field lists the organizations available in the `lms` database, while also allowing users to add in an organization that is not available in the list as a string value.

## Changes Made

- Implemented necessary backend modifications to store and process the organization data.
- Updated the User Profile model to accommodate the new field that will store the organization field as a string.
- Migrated all relevant changes.
- Created a new GET API that retrieves the list of organizations to populate the dropdown at the frontend. Accessible at [http://localhost:18000/api/user/v1/organizations/registration](http://localhost:18000/api/user/v1/organizations/registration)
- Wrote a new testing class for the new GET API mentioned above (all tests pass).
- Wrote validation functions for the new field to ensure the field follows the specified format and length requirements.

## How to Test

1. Load the Organization data fixture into lms using the following command `python manage.py lms loaddata common/test/db_fixtures/edx-organizations.json --app organizations.organization` on the terminal for lms.
2. Make sure lms is running at `localhost:18000`
3. Access the registration form on the [website](http://localhost:1999/register) and register a new user.

## Issues / Improvements

1. Change the data type of the "organization" attribute in the User Profile model from a `string` to `Organization` so that it stores an actual object of type Organization. This update will enhance the integrity and consistency of the data stored within the model.
2. Modify the API's functionality to verify the validity of the submitted `organization's primary key` by cross-checking it against the existing database records. Allow successful registration only if the key matches an entry in the database, ensuring that only valid organization names are stored.

## Checklist

- [ ] The code follows our established coding guidelines.
- [ ] Unit tests have been added to ensure the proper functionality of the new field.
- [ ] Documentation has been updated to reflect the changes made.

Please review and provide feedback. Thank you.
